front/components/agent_builder/AgentBuilderPreview.tsx

```diff
@@ -51,33 +52,33 @@ function LoadingState({ message }: LoadingStateProps) {
 export function AgentBuilderPreview() {
   const { owner } = useAgentBuilderContext();
   const { user } = useUser();
+  const { getValues } = useFormContext<AgentBuilderFormData>();
   const { isMCPServerViewsLoading } = useMCPServerViewsContext();
 
-  const instructions = useWatch<AgentBuilderFormData, "instructions">({
-    name: "instructions",
-  });
-  const name = useWatch<AgentBuilderFormData, "agentSettings.name">({
-    name: "agentSettings.name",
-  });
+  const hasContent =
+    !!getValues("instructions").trim() || getValues("actions").length > 0;
 
   const {
     draftAgent,
+    getDraftAgent,
     isSavingDraftAgent,
     draftCreationFailed,
-    createDraftAgent,
-  } = usePreviewAgent();
-  const { conversation, stickyMentions, setStickyMentions, handleSubmit } =
-    useTryAgentCore({
-      owner,
-      user,
-      agent: draftAgent,
-      createDraftAgent,
-    });
-
-  const hasContent = instructions.trim();
+    stickyMentions,
+    setStickyMentions,
+  } = useDraftAgent();
+
+  const { conversation, handleSubmit } = useDraftConversation({
+    draftAgent,
+    getDraftAgent,
+  });
+
+  // Show loading spinner only when the first time we create a draft agent. After that the spinner is shown
+  // inside the button in the input bar. This way we don't have to unmount the conversation viewer every time.
+  const showLoader =
+    !draftAgent && (isMCPServerViewsLoading || isSavingDraftAgent);
```

Comment:
I have a flickering issue when:
- creating a new agent
- starting writing in the instructions block

We show the input bar, then it get hidden and we display a spinner, then we show the input bar again